RecoTracker/LSTCore/src/alpaka/Triplet.h

```diff
@@ -0,0 +1,1125 @@
+                                                                struct SDL::miniDoublets& mdsInGPU,
+                                                                struct SDL::segments& segmentsInGPU,
+                                                                uint16_t& innerInnerLowerModuleIndex,
+                                                                uint16_t& middleLowerModuleIndex,
+                                                                uint16_t& outerOuterLowerModuleIndex,
+                                                                unsigned int& firstMDIndex,
+                                                                unsigned int& secondMDIndex,
+                                                                unsigned int& thirdMDIndex,
+                                                                float& zOut,
+                                                                float& rtOut,
+                                                                uint16_t& innerOuterLowerModuleIndex,
+                                                                unsigned int& innerSegmentIndex,
+                                                                unsigned int& outerSegmentIndex,
+                                                                float& betaIn,
+                                                                float& betaInCut) {
+    bool pass = true;
+
+    bool isPSIn = (modulesInGPU.moduleType[innerInnerLowerModuleIndex] == SDL::PS);
+    bool isPSOut = (modulesInGPU.moduleType[outerOuterLowerModuleIndex] == SDL::PS);
+
+    float rtIn = mdsInGPU.anchorRt[firstMDIndex];
+    float rtMid = mdsInGPU.anchorRt[secondMDIndex];
+    rtOut = mdsInGPU.anchorRt[thirdMDIndex];
+
+    float zIn = mdsInGPU.anchorZ[firstMDIndex];
+    float zMid = mdsInGPU.anchorZ[secondMDIndex];
+    zOut = mdsInGPU.anchorZ[thirdMDIndex];
+
+    float alpha1GeV_OutLo =
+        alpaka::math::asin(acc, alpaka::math::min(acc, rtOut * SDL::k2Rinv1GeVf / SDL::ptCut, SDL::sinAlphaMax));
+
+    float dzDrtScale =
+        alpaka::math::tan(acc, alpha1GeV_OutLo) / alpha1GeV_OutLo;  // The track can bend in r-z plane slightly
+    float zpitchIn = (isPSIn ? SDL::pixelPSZpitch : SDL::strip2SZpitch);
+    float zpitchOut = (isPSOut ? SDL::pixelPSZpitch : SDL::strip2SZpitch);
+    float zGeom = zpitchIn + zpitchOut;
+
+    // Cut #0: Preliminary (Only here in endcap case)
+    pass = pass and (zIn * zOut > 0);
+    if (not pass)
+      return pass;
+
+    float dLum = SDL::copysignf(SDL::deltaZLum, zIn);
+    bool isOutSgInnerMDPS = modulesInGPU.moduleType[outerOuterLowerModuleIndex] == SDL::PS;
+    float rtGeom1 = isOutSgInnerMDPS ? SDL::pixelPSZpitch : SDL::strip2SZpitch;
+    float zGeom1 = SDL::copysignf(zGeom, zIn);
+    float rtLo = rtIn * (1.f + (zOut - zIn - zGeom1) / (zIn + zGeom1 + dLum) / dzDrtScale) -
+                 rtGeom1;  //slope correction only on the lower end
+    zOut = zOut;
+    rtOut = rtOut;
```

Comment:
```
src/RecoTracker/LSTCore/src/alpaka/Triplet.h:454:10: warning: explicitly assigning value of variable of type 'float' to itself [-Wself-assign]
	zOut = zOut;
	~~~~ ^ ~~~~
src/RecoTracker/LSTCore/src/alpaka/Triplet.h:455:11: warning: explicitly assigning value of variable of type 'float' to itself [-Wself-assign]
	rtOut = rtOut;
	~~~~~ ^ ~~~~~
```
What are we doing exactly here?

Reply:
Yeah, this needs to be fixed. This warning only started appearing when we started compiling for ROCm, so it wasn't fixed when Slava cleaned up all warnings.